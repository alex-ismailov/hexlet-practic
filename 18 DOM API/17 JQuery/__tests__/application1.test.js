import fs from 'fs';
import path from 'path';
import { html } from 'js-beautify';

import run from '../src/application.js';

const fixuturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => html(document.body.innerHTML);

beforeAll(() => {
  const initHtml = fs.readFileSync(path.join(fixuturesPath, 'index1.html')).toString();
  document.documentElement.innerHTML = initHtml;
  run();
});

test('application', () => {
  const prev = document.querySelector('a[data-slide="prev"]');
  const next = document.querySelector('a[data-slide="next"]');
  next.click();
  expect(getTree()).toMatchSnapshot();

  prev.click();
  expect(getTree()).toMatchSnapshot();

  next.click();
  expect(getTree()).toMatchSnapshot();

  next.click();
  expect(getTree()).toMatchSnapshot();

  next.click();
  expect(getTree()).toMatchSnapshot();

  next.click();
  expect(getTree()).toMatchSnapshot();

  prev.click();
  expect(getTree()).toMatchSnapshot();

  prev.click();
  expect(getTree()).toMatchSnapshot();
});
