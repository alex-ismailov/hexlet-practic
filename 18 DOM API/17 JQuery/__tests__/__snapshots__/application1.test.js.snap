// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1`] = `
"<div class=\\"container m-3\\">
    <div id=\\"carouselExampleControls\\" class=\\"carousel slide\\" data-ride=\\"carousel\\">
        <div class=\\"carousel-inner\\" role=\\"listbox\\">
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=First slide\\" alt=\\"First slide\\">
            </div>
            <div class=\\"carousel-item active\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Second slide\\" alt=\\"Second slide\\">
            </div>
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Third slide\\" alt=\\"Third slide\\">
            </div>
        </div>
        <a class=\\"carousel-control-prev\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"prev\\">
            <span class=\\"carousel-control-prev-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Previous</span>
        </a>
        <a class=\\"carousel-control-next\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"next\\">
            <span class=\\"carousel-control-next-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Next</span>
        </a>
    </div>
</div>"
`;

exports[`application 2`] = `
"<div class=\\"container m-3\\">
    <div id=\\"carouselExampleControls\\" class=\\"carousel slide\\" data-ride=\\"carousel\\">
        <div class=\\"carousel-inner\\" role=\\"listbox\\">
            <div class=\\"carousel-item active\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=First slide\\" alt=\\"First slide\\">
            </div>
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Second slide\\" alt=\\"Second slide\\">
            </div>
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Third slide\\" alt=\\"Third slide\\">
            </div>
        </div>
        <a class=\\"carousel-control-prev\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"prev\\">
            <span class=\\"carousel-control-prev-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Previous</span>
        </a>
        <a class=\\"carousel-control-next\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"next\\">
            <span class=\\"carousel-control-next-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Next</span>
        </a>
    </div>
</div>"
`;

exports[`application 3`] = `
"<div class=\\"container m-3\\">
    <div id=\\"carouselExampleControls\\" class=\\"carousel slide\\" data-ride=\\"carousel\\">
        <div class=\\"carousel-inner\\" role=\\"listbox\\">
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=First slide\\" alt=\\"First slide\\">
            </div>
            <div class=\\"carousel-item active\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Second slide\\" alt=\\"Second slide\\">
            </div>
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Third slide\\" alt=\\"Third slide\\">
            </div>
        </div>
        <a class=\\"carousel-control-prev\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"prev\\">
            <span class=\\"carousel-control-prev-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Previous</span>
        </a>
        <a class=\\"carousel-control-next\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"next\\">
            <span class=\\"carousel-control-next-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Next</span>
        </a>
    </div>
</div>"
`;

exports[`application 4`] = `
"<div class=\\"container m-3\\">
    <div id=\\"carouselExampleControls\\" class=\\"carousel slide\\" data-ride=\\"carousel\\">
        <div class=\\"carousel-inner\\" role=\\"listbox\\">
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=First slide\\" alt=\\"First slide\\">
            </div>
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Second slide\\" alt=\\"Second slide\\">
            </div>
            <div class=\\"carousel-item active\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Third slide\\" alt=\\"Third slide\\">
            </div>
        </div>
        <a class=\\"carousel-control-prev\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"prev\\">
            <span class=\\"carousel-control-prev-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Previous</span>
        </a>
        <a class=\\"carousel-control-next\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"next\\">
            <span class=\\"carousel-control-next-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Next</span>
        </a>
    </div>
</div>"
`;

exports[`application 5`] = `
"<div class=\\"container m-3\\">
    <div id=\\"carouselExampleControls\\" class=\\"carousel slide\\" data-ride=\\"carousel\\">
        <div class=\\"carousel-inner\\" role=\\"listbox\\">
            <div class=\\"carousel-item active\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=First slide\\" alt=\\"First slide\\">
            </div>
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Second slide\\" alt=\\"Second slide\\">
            </div>
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Third slide\\" alt=\\"Third slide\\">
            </div>
        </div>
        <a class=\\"carousel-control-prev\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"prev\\">
            <span class=\\"carousel-control-prev-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Previous</span>
        </a>
        <a class=\\"carousel-control-next\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"next\\">
            <span class=\\"carousel-control-next-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Next</span>
        </a>
    </div>
</div>"
`;

exports[`application 6`] = `
"<div class=\\"container m-3\\">
    <div id=\\"carouselExampleControls\\" class=\\"carousel slide\\" data-ride=\\"carousel\\">
        <div class=\\"carousel-inner\\" role=\\"listbox\\">
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=First slide\\" alt=\\"First slide\\">
            </div>
            <div class=\\"carousel-item active\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Second slide\\" alt=\\"Second slide\\">
            </div>
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Third slide\\" alt=\\"Third slide\\">
            </div>
        </div>
        <a class=\\"carousel-control-prev\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"prev\\">
            <span class=\\"carousel-control-prev-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Previous</span>
        </a>
        <a class=\\"carousel-control-next\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"next\\">
            <span class=\\"carousel-control-next-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Next</span>
        </a>
    </div>
</div>"
`;

exports[`application 7`] = `
"<div class=\\"container m-3\\">
    <div id=\\"carouselExampleControls\\" class=\\"carousel slide\\" data-ride=\\"carousel\\">
        <div class=\\"carousel-inner\\" role=\\"listbox\\">
            <div class=\\"carousel-item active\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=First slide\\" alt=\\"First slide\\">
            </div>
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Second slide\\" alt=\\"Second slide\\">
            </div>
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Third slide\\" alt=\\"Third slide\\">
            </div>
        </div>
        <a class=\\"carousel-control-prev\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"prev\\">
            <span class=\\"carousel-control-prev-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Previous</span>
        </a>
        <a class=\\"carousel-control-next\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"next\\">
            <span class=\\"carousel-control-next-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Next</span>
        </a>
    </div>
</div>"
`;

exports[`application 8`] = `
"<div class=\\"container m-3\\">
    <div id=\\"carouselExampleControls\\" class=\\"carousel slide\\" data-ride=\\"carousel\\">
        <div class=\\"carousel-inner\\" role=\\"listbox\\">
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=First slide\\" alt=\\"First slide\\">
            </div>
            <div class=\\"carousel-item\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Second slide\\" alt=\\"Second slide\\">
            </div>
            <div class=\\"carousel-item active\\">
                <img class=\\"d-block img-fluid\\" src=\\"holder.js/1000x400?auto=yes&amp;bg=555&amp;fg=333&amp;text=Third slide\\" alt=\\"Third slide\\">
            </div>
        </div>
        <a class=\\"carousel-control-prev\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"prev\\">
            <span class=\\"carousel-control-prev-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Previous</span>
        </a>
        <a class=\\"carousel-control-next\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"next\\">
            <span class=\\"carousel-control-next-icon\\" aria-hidden=\\"true\\"></span>
            <span class=\\"sr-only\\">Next</span>
        </a>
    </div>
</div>"
`;
